### PR TITLE
docs: add `repo:contains`

### PR DESCRIPTION
**Pull request by rvantonder**
This updates the language doc, and OpenGrok doc. 

Updates look a bit out of place in the the tabular search query docs, because we don't support negation. Will see if I can PR something like that soon and will update then. The partial update here is good enough to ship.

 I will also remove the Deprecated note one iteration from now.

### CODE REVIEW

File: doc/code_search/reference/language.md [hunk 1]
```diff
@@ -196,31 +196,33 @@ Search parameters allow you to filter search results or modify search behavior.
...
+		Choice(0,
+			Skip(),
+			Terminal("-"),
+			Sequence(
+				Terminal("NOT"),
+				Terminal("space", {href: "#whitespace"}))),
+		Choice(0,
+			Terminal("repo:"),
+			Terminal("r:")),
+		Choice(0,
```
Comment (camdencheek) [inline doc/code_search/reference/language.md:208]: nit: indentation is a bit funky here which makes it difficult to read. Is likely an issue with my editor config from making this in the first place -- sorry 😄

File: doc/code_search/reference/language.md [hunk 2]
```diff
@@ -501,7 +503,7 @@ ComplexDiagram(
     Terminal("regular expression", {href: "#regular-expression"})).addTo();
 </script>
 
-Only include results from repositories that contain a matching file. This
+_Deprecated. Prefer [Repo contains file](#repo-contains-file)._ Only include results from repositories that contain a matching file. This
```
Comment (camdencheek) [inline doc/code_search/reference/language.md:506]: 🎉

File: doc/code_search/reference/language.md [hunk 3]
```diff
@@ -599,6 +602,78 @@ ComplexDiagram(
 
 A string that is interpreted as a <a href="https://golang.org/s/re2syntax">RE2</a> regular expression.
 
+## Built-in predicate
```
Comment (camdencheek) [inline doc/code_search/reference/language.md:605]: I expect we'll have predicates for non-`repo` fields in the near future. Should we clarify that these are predicates specifically for the `repo:` field?

File: doc/code_search/reference/language.md [hunk 4]
```diff
@@ -599,6 +602,78 @@ ComplexDiagram(
...
+
+<script>
+ComplexDiagram(
+    Choice(0,
+        Terminal("contains.content(...)", {href: "#repo-contains-content"}),
+        Terminal("contains.file(...)", {href: "#contains-file"}),
+        Terminal("contains(...)", {href: "#contains"}),
```
Comment (camdencheek) [inline doc/code_search/reference/language.md:612]: Looks to me like `#contains` is a dead ref

**Review by camdencheek — APPROVED** [review]

File: doc/code_search/reference/language.md (hunk 3 above)
Comment (rvantonder) [reply doc/code_search/reference/language.md:593]: I figured to segment that out once we do have them for non-repo. It's probably at least a month or more away

File: doc/code_search/reference/language.md (hunk 2 above)
Comment (rvantonder) [reply doc/code_search/reference/language.md:506]: Actually feel like this is a bit premature given https://github.com/sourcegraph/sourcegraph/issues/20337 but it's OK

File: doc/code_search/reference/language.md (hunk 4 above)
Comment (rvantonder) [reply doc/code_search/reference/language.md]: Good eye!

**Pull request merged**